src/utils/requestUtil.ts

```diff
@@ -4,16 +4,37 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { WebResource } from "ms-rest";
+import { ServiceClientCredentials } from "ms-rest";
 import * as request from 'request-promise';
 import { appendExtensionUserAgent } from "vscode-azureextensionui";
+import { IFunctionHostKeyResponse } from "../models/Function";
+import { signRequest } from "./signRequest";
 
-export async function requestUtil(url: string) : Promise<string> {
+export type nRequest = WebResource & request.RequestPromiseOptions;
+
+export async function requestUtil(url: string): Promise<string> {
     const requestOptions: WebResource = new WebResource();
     requestOptions.headers = {
         ['User-Agent']: appendExtensionUserAgent()
     };
     requestOptions.url = url;
-// tslint:disable-next-line: await-promise
+    // tslint:disable-next-line: await-promise
     const response = await request(requestOptions).promise();
-    return  <string>(response);
+    return <string>(response);
+}
+
+export async function requestUtilWithCredentials(url: string, credentials: ServiceClientCredentials): Promise<string> {
```

Comment:
Do we need another function or just use the previous function and make the param credentials: ServiceClientCredentials optional